packages/core/saved-objects/core-saved-objects-migration-server-internal/src/kibana_migrator.ts

```diff
@@ -37,6 +37,10 @@ import { createIndexMap } from './core/build_index_map';
 import { runResilientMigrator } from './run_resilient_migrator';
 import { migrateRawDocsSafely } from './core/migrate_raw_docs';
 
+// ensure plugins don't try to convert SO namespaceTypes after 8.0.0
+// see https://github.com/elastic/kibana/issues/147344
+const ALLOWED_CONVERT_VERSION = '8.0.0';
```

Comment:
I'll defer to @rudolf, but are we sure we don't want to allow type owners to migrate to shareable until some versions to come? This PR basically hard block any type from being converted as soon as merged. I want to make sure we're fine with the consequences.

Reply:
if I understood correctly, this is a pre-requisite to [only migrate an index if necessary](https://github.com/elastic/kibana/issues/124946).

Reply:
FWIW I just saw https://github.com/elastic/kibana/pull/147198/, but it is converting from "agnostic".

Reply:
My 2cents are: as claimed in https://github.com/elastic/kibana/issues/147344, if we are not reindexing when there are no mapping updates (or, at least, no breaking changes), we cannot support any further version unless we change the current logic to identify that we need reindex anyway because a conversion must happen.

WDYT?

Reply:
yeah if someone REALLY REALLY REALLY needs this we'll need to re-enable and detect this to ensure we reindex (which again will introduce downtime).

Re https://github.com/elastic/kibana/pull/147198 this is sort of a unique case because it's a new saved object type, there aren't any `files` saved objects in users clusters.